cpp/ql/src/Likely Bugs/Memory Management/ReturnStackAllocatedMemory.ql

```diff
@@ -75,14 +75,23 @@ class ReturnStackAllocatedMemoryConfig extends MustFlowConfiguration {
   }
 }
 
+predicate instrHasEnclosingCallable(VariableAddressInstruction var, Function f) {
+  f = var.getEnclosingFunction()
+}
+
+predicate nodeHasEnclosingCallable(DataFlow::Node node, Function f) {
+  f = node.getEnclosingCallable()
+}
+
 from
   MustFlowPathNode source, MustFlowPathNode sink, VariableAddressInstruction var,
-  ReturnStackAllocatedMemoryConfig conf
+  ReturnStackAllocatedMemoryConfig conf, Function f
 where
   conf.hasFlowPath(source, sink) and
   source.getNode().asInstruction() = var and
   // Only raise an alert if we're returning from the _same_ callable as the on that
   // declared the stack variable.
-  var.getEnclosingFunction() = sink.getNode().getEnclosingCallable()
+  instrHasEnclosingCallable(var, pragma[only_bind_into](f)) and
+  nodeHasEnclosingCallable(sink.getNode(), pragma[only_bind_into](f))
```

Comment:
Since you're doing this. I assume you cannot do something like:
```
var.getEnclosingFunction() = pragma[only_bind_into](f) and
sink.getNode().getEnclosingCallable() = pragma[only_bind_into](f)
```
Or does that have a different effect?

Reply:
I personally find that a bit cleaner, as each predicate is only a single line, but I'll leave the choice to you.

Reply:
I would also prefer not to have the extra predicates if they're unnecessary (but if they are, I'm OK with the code as it is).

Reply:
In that case, I'll remove the added predicates. That also automatically resolves https://github.com/github/codeql/pull/8427#discussion_r825830021 🎉.